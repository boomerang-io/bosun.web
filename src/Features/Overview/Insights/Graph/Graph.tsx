import React, { Component } from "react";
import { Tile } from "@boomerang-io/carbon-addons-boomerang-react";
import LineChart from "./LineChart";
import { ChartsData } from "Types";
import styles from "./graph.module.scss";

type Props = {
  formatedData: ChartsData;
};

export class Graph extends Component<Props> {
  state = {
    fromDate: {},
    standard: {},
    toDate: {},
  };

  render() {
    const { formatedData } = this.props;
    return (
      <Tile
        style={{
          display: "flex",
          flexDirection: "column",
          alignItems: "spaceBetween",
          minHeight: "10rem",
        }}
      >
        <div className={styles.header}>
          <h3 className={styles.title}>Violations Trend</h3>
          <div className={styles.filters}></div>
        </div>
        <div className={styles.container} data-testid="insights-graph">
          {formatedData.chartData.length > 0 ? (
            <LineChart
              chartData={formatedData.chartData}
              lines={formatedData.lines}
              higherValue={formatedData.higherValue}
            />
          ) : (
            <p>No violations data to be found</p>
          )}
        </div>
      </Tile>
    );
  }
}

export default Graph;
