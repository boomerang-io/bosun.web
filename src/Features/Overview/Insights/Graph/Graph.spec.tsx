import React from "react";
import Graph from "./index";

const formatedData = {
  chartData: [{ date: 1558457676, test1: 1, test2: 2 }],
  lines: ["test1", "test2"],
  higherValue: 2,
  totalViolations: 0,
};

describe("Graph --- Snapshot", () => {
  it("Capturing Snapshot of Graph", async () => {
    //TO-DO Check how to handle this kind of object
    //@ts-ignore
    const { baseElement } = (global as any).rtlRender(<Graph formatedData={formatedData} />);
    expect(baseElement).toMatchSnapshot();
  });
});
