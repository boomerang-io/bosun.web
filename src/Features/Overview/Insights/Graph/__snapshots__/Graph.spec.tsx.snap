// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graph --- Snapshot Capturing Snapshot of Graph 1`] = `
<body
  id="app"
>
  <div>
    <div
      class="bx--tile"
      style="display: flex; flex-direction: column; align-items: spaceBetween; min-height: 10rem;"
    >
      <div
        class="header"
      >
        <h3
          class="title"
        >
          Violations Trend
        </h3>
        <div
          class="filters"
        />
      </div>
      <div
        class="container"
        data-testid="insights-graph"
      >
        <div
          style="display: flex; align-items: center; width: 100%;"
        >
          <div
            class="recharts-responsive-container"
            style="width: 99%; height: 100%;"
          >
            <div
              style="position: absolute; width: 0px; height: 0px; visibility: hidden; display: none;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
